homeassistant/components/nest/camera_sdm.py

```diff
@@ -156,7 +163,40 @@ async def async_added_to_hass(self):
 
     async def async_camera_image(self):
         """Return bytes of camera image."""
+        # Returns the snapshot of the last event for ~30 seconds after the event
+        active_event_image = await self._async_active_event_image()
+        if active_event_image:
+            return active_event_image
+        # Fetch still image from the live stream
         stream_url = await self.stream_source()
         if not stream_url:
             return None
         return await async_get_image(self.hass, stream_url, output_format=IMAGE_JPEG)
+
+    async def _async_active_event_image(self):
+        """Return image from any active events happening."""
+        if CameraEventImageTrait.NAME not in self._device.traits:
+            return None
+        trait = self._device.active_event_trait
+        if not trait:
+            return None
+        # Reuse image bytes if they have already been fetched
+        event_id = trait.last_event.event_id
+        if self._event_id is not None and self._event_id == event_id:
+            return self._event_image_bytes
+        _LOGGER.info("Fetching URL for event_id %s", event_id)
+        try:
+            event_image = await trait.generate_active_event_image()
+        except GoogleNestException as err:
+            _LOGGER.debug("Unable to generate event image URL: %s", err)
+            return None
+        if not event_image:
+            return None
+        try:
+            image_bytes = await event_image.contents()
+        except GoogleNestException as err:
+            _LOGGER.debug("Unable to fetch event image: %s", err)
+            return None
+        self._event_id = event_id
+        self._event_image_bytes = image_bytes
```

Comment:
Should we add some automatic clean up here to make sure it is cleared from memory after it expires ?

Reply:
My initial thought was that this would not be very larger, say, a few hundred KB at most which I assume isn't a problem for even small devices.   From a privacy perspective, i think it makes sense and clearing it would be in the spirit of the API URL expiration.  This is not hard so let me send a follow up to do this. I'll think about if its better to have separate alarms or reuse the existing alarm.

Reply:
Also just to make sure it is clear for history, the event id check is there to ensure correctness and that a stale image is not shown.